cmd/skopeo/proxy.go

```diff
@@ -568,6 +573,52 @@ func (h *proxyHandler) GetBlob(args []interface{}) (replyBuf, error) {
 	return ret, nil
 }
 
+func (h *proxyHandler) LayerInfoJSON(args []interface{}) (replyBuf, error) {
+  h.lock.Lock()
+	defer h.lock.Unlock()
+
+	var ret replyBuf
+
+	if h.sysctx == nil {
+		return ret, fmt.Errorf("client error: must invoke Initialize")
+	}
+	
+  if len(args) != 1 {
+		return ret, fmt.Errorf("found %d args, expecting (imgid)", len(args))
+	}
+  
+  imgref, err := h.parseImageFromID(args[0])
+  if err != nil {
+    return ret, err
+  }
+
+  ctx := context.TODO()
+
+  var infos []types.BlobInfo
+
+  err = h.cacheTargetManifest(imgref)
+	if err != nil {
+		return ret, err
+	}
+	img := imgref.cachedimg
+
+  infos, err = img.LayerInfosForCopy(ctx)
+  if err != nil {
+    return ret, err
+  }
+  
+  if infos == nil {
+     infos = img.LayerInfos()
+  }
+
+  var serStruct layersInfo
+  serStruct.Success = "successfull"
```

Comment:
I think we can drop this; we already have an error reporting mechanism.  Then we can just do `ret.value = infos` which would seem to simplify things.